src/daft-connect/src/translation/expr/unresolved_function.rs

```diff
@@ -96,7 +98,13 @@ pub fn handle_count(arguments: Vec<daft_dsl::ExprRef>) -> eyre::Result<daft_dsl:
 
     let [arg] = arguments;
 
-    let count = arg.count(CountMode::All);
+    let arg = if arg.as_literal().and_then(|lit| lit.as_i32()) == Some(1i32) {
+        col("*")
```

Comment:
in spark, `df.count()` is processed as `count(1)`, which is equivalent to our `count(*)`